src/qibosoq/programs/base.py

```diff
@@ -318,7 +310,7 @@ def declare_gen_mux_ro(self):
             ro_ch=adc_ch_added[0],
         )
 
-    def add_muxed_readout_to_register(self, ro_pulses: List[Pulse]):
+    def add_muxed_readout_to_register(self, ro_pulses: List[Rectangular]):
```

Comment:
Again, just to understand myself: do you know why multiplexing is limited to rectangular pulses?

Reply:
If I remember correctly this is a limitation of qick (I don;t know if software or firmware). 
I think the "set_pulse_register" function, where you set which waveform to use, doesn't work completly... But I would have to check

Reply:
Thanks for the answer, no current need for a deeper investigation